qiskit_experiments/calibration/experiments/spectroscopy.py

```diff
@@ -0,0 +1,284 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy experiment class."""
+
+from typing import List, Optional, Tuple, Union
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Gate
+import qiskit.pulse as pulse
+from qiskit.qobj.utils import MeasLevel
+from qiskit_experiments.analysis.curve_fitting import curve_fit
+from qiskit_experiments.base_analysis import BaseAnalysis
+from qiskit_experiments.base_experiment import BaseExperiment
+from qiskit_experiments import AnalysisResult
+from qiskit_experiments.data_processing.data_processor import DataProcessor
+from qiskit_experiments.data_processing.nodes import ToReal
+from qiskit_experiments.data_processing.nodes import Probability
+
+
+class SpectroscopyAnalysis(BaseAnalysis):
+    """Class to analysis a spectroscopy experiment."""
+
+    # pylint: disable=arguments-differ
+    def _run_analysis(
+        self,
+        experiment_data,
+        data_processor=None,
+        meas_level=MeasLevel.KERNELED,
+        amp_guess: float = None,
+        gamma_guesses: List[float] = None,
+        freq_guess: float = None,
```

Comment:
I'm not familiar with the nuances of type hints. I think this should be `Optional[float] = None` (here and in other places).  https://stackoverflow.com/questions/51710037/how-should-i-use-the-optional-type-hint

Reply:
Super, thanks a lot for the link. See here 1f19f3d .